processing/src/main/java/org/apache/druid/math/expr/Function.java

```diff
@@ -3724,6 +3724,44 @@ ExprEval doApply(ExprEval arrayExpr, ExprEval scalarExpr)
     }
   }
 
+  class ArrayScalarInFunction extends ArrayScalarFunction
+  {
+    @Override
+    public String name()
+    {
+      return "scalar_in";
+    }
+
+    @Nullable
+    @Override
+    public ExpressionType getOutputType(Expr.InputBindingInspector inspector, List<Expr> args)
+    {
+      return ExpressionType.LONG;
+    }
+
+    @Override
+    Expr getScalarArgument(List<Expr> args)
+    {
+      return args.get(0);
+    }
+
+    @Override
+    Expr getArrayArgument(List<Expr> args)
+    {
+      return args.get(1);
+    }
+
+    @Override
+    ExprEval doApply(ExprEval arrayExpr, ExprEval scalarExpr)
```

Comment:
You don't need to do this now, but as a follow up you could add an `asSingleThreaded` impl that creates a Set to avoid the `Arrays.asList(array).contains`, similar to the technique used in `OverlapConstantArray`.